### PR TITLE
avm2: Implement DisplayObject.blendShader

### PR DESCRIPTION
**Pull request by Aaron1011**
We already had almost all of the PixelBender support needed for this. Since this is run during displayobject rendering, we need to support MSAA, which requires us to create a `RenderPipeline` per sample count used.

This fixes Closure's rendering (previously, the game rendered as a solid white background with a few gray ovals).

### CODE REVIEW

**Comment by Lord-McSweeney** [comment]
This also closes #58.

File: render/wgpu/src/pixel_bender.rs [hunk 1]
```diff
@@ -39,6 +43,57 @@ pub struct PixelBenderWgpuShader {
...
+                            entry_point: naga_pixelbender::VERTEX_SHADER_ENTRYPOINT,
+                            buffers: &VERTEX_BUFFERS_DESCRIPTION_FILTERS,
+                        },
+                        fragment: Some(wgpu::FragmentState {
+                            module: &self.fragment_shader,
+                            entry_point: naga_pixelbender::FRAGMENT_SHADER_ENTRYPOINT,
+                            targets: &[Some(ColorTargetState {
+                                format: TextureFormat::Rgba8Unorm,
+                                // FIXME - what should this be?
+                                blend: Some(wgpu::BlendState {
```
Comment (Dinnerbone) [inline render/wgpu/src/pixel_bender.rs:72]: `None`, I imagine? Unless we're expected to blend the blend result.

File: render/wgpu/src/pixel_bender.rs [hunk 2]
```diff
@@ -29,7 +30,10 @@ use crate::{
 #[derive(Debug)]
 pub struct PixelBenderWgpuShader {
     bind_group_layout: wgpu::BindGroupLayout,
-    pipeline: wgpu::RenderPipeline,
+    pipeline_layout: PipelineLayout,
+    pipelines: RefCell<FnvHashMap<u32, RenderPipeline>>,
```
Comment (Dinnerbone) [inline render/wgpu/src/pixel_bender.rs:34]: `SampleCountMap` exists for this

File: swf/src/types.rs [hunk 3]
```diff
@@ -387,6 +387,8 @@ pub enum BlendMode {
     Erase = 12,
     Overlay = 13,
     HardLight = 14,
+    // FIXME - prevent this from being read from SWF
+    Shader = 15,
```
Comment (Dinnerbone) [inline swf/src/types.rs:391]: For cases like these, I think we should have a similar (but "enhanced") enum in ruffle_render, which has the new types that a swf can't know about. It feels like a big hack to have these exist in a serialization crate.

File: render/wgpu/src/pixel_bender.rs (hunk 1 above)
Comment (Aaron1011) [reply render/wgpu/src/pixel_bender.rs:72]: This is a pre-existing FIXME - I just moved some code around

Comment (Dinnerbone) [reply render/wgpu/src/pixel_bender.rs:72]: Got it!

**Review by Dinnerbone — APPROVED** [review]

**Pull request merged**